htmlutil/htmlutil.go

```diff
@@ -0,0 +1,112 @@
+package htmlutil
+
+import (
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+)
+
+// CreateNode creates an html Node and sets attributes or adds child nodes according to the type of each value
+func CreateNode(data string, dataAtom atom.Atom, values ...interface{}) *html.Node {
+	node := &html.Node{
+		Type:     html.ElementNode,
+		Data:     data,
+		DataAtom: dataAtom,
+	}
+	for _, value := range values {
+		switch v := value.(type) {
+		case html.Attribute:
+			node.Attr = append(node.Attr, v)
+		case *html.Node:
+			node.AppendChild(v)
+		case []*html.Node:
+			for _, c := range v {
+				node.AppendChild(c)
+			}
+		case string:
+			node.AppendChild(&html.Node{Type: html.TextNode, Data: v})
+		}
+	}
+	return node
+}
+
+// AddAttribute adds an attribute to the node
+func AddAttribute(node *html.Node, key string, val string) {
+	node.Attr = append(node.Attr, html.Attribute{Key: key, Val: val})
+}
+
+// ReplaceAttribute adds an attribute to the node, replacing any existing attribute with the same name
+func ReplaceAttribute(node *html.Node, key string, val string) {
+	var attr []html.Attribute
+	for _, a := range node.Attr {
+		if a.Key != key {
+			attr = append(attr, a)
+		}
+	}
+	node.Attr = append(attr, html.Attribute{Key: key, Val: val})
+}
+
+// Attr creates a new Attribute
+func Attr(key string, val string) html.Attribute {
+	return html.Attribute{Key: key, Val: val}
+}
+
+// Text creates a new Text node
+func Text(text string) *html.Node {
+	return &html.Node{Type: html.TextNode, Data: text}
+}
+
+// GetAttribute finds an attribute for the node - returns empty string if not found
+func GetAttribute(node *html.Node, key string) string {
+	for _, attr := range node.Attr {
+		if attr.Key == key {
+			return attr.Val
+		}
+	}
```

Comment:
is it worth logging here the attribute could not be found?

Reply:
GetAttribute is used by HasAttributes, so you'd end up with a *lot* of debug output